### PR TITLE
Created AbstractStreamTable to make it easy to create custom variants

### PR DESCRIPTION
**Pull request by ham1**
OptimisedStreamTable is the same as StreamTable but without checking
each element for isDisplayed which dramatically speeds it up, but comes
at a cost of not coping with Tables with hidden columns or rows.
Added tests demonstrating two different approaches to dealing with tables.

### CODE REVIEW

File: src/main/java/com/frameworkium/core/ui/element/AbstractStreamTable.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    public Optional<WebElement> getHeading(Predicate<WebElement> headerMatcher) {
+        return getHeading(getHeaderIndex(headerMatcher));
+    }
+
+    /**
+     * @return {@link Stream} of {@link Stream} of {@code td} {@link WebElement}s
+     */
+    public Stream<Stream<WebElement>> getRows() {
+        return rows()
+                .map(el -> el.findElements(By.cssSelector("td")).stream());
```
Comment (Mjl33) [inline src/main/java/com/frameworkium/core/ui/element/AbstractStreamTable.java]: Custom variants would have "td" customized as well

File: src/main/java/com/frameworkium/core/ui/element/OptimisedStreamTable.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    @FindBy(css = "thead > tr > th")
+    private List<WebElement> headerCells;
+
+    @FindBy(css = "tbody > tr")
+    private List<WebElement> rows;
+
+    @Override
+    protected Stream<WebElement> headerCells() {
+        return headerCells.stream().filter(WebElement::isDisplayed);
```
Comment (Mjl33) [inline src/main/java/com/frameworkium/core/ui/element/OptimisedStreamTable.java]: is isDisplayed check here appropriate for the optimized version?

File: src/main/java/com/frameworkium/core/ui/element/AbstractStreamTable.java (hunk 1 above)
Comment (ham1) [reply src/main/java/com/frameworkium/core/ui/element/AbstractStreamTable.java]: Agreed.

File: src/main/java/com/frameworkium/core/ui/element/OptimisedStreamTable.java (hunk 2 above)
Comment (ham1) [reply src/main/java/com/frameworkium/core/ui/element/OptimisedStreamTable.java]: Copy-paste mistake, already fixed. :)

File: src/main/java/com/frameworkium/core/ui/element/AbstractStreamTable.java [hunk 3]
```diff
@@ -217,13 +212,13 @@
...
-        return IntStream.range(0, headings.size())
-                .filter(i -> headerPredicate.test(headings.get(i)))
+    private long getHeaderIndex(Predicate<WebElement> headerPredicate) {
+        return Streams.mapWithIndex(
+                getHeadings(), (we, i) -> headerPredicate.test(we) ? i : null)
+                .filter(Objects::nonNull)
                 .findFirst()
                 .orElseThrow(() -> new NoSuchElementException(
-                        "Table header " + headerPredicate + " not found"));
+                        "No header matches " + headerPredicate));
```
Comment (Mjl33) [inline src/main/java/com/frameworkium/core/ui/element/AbstractStreamTable.java]: I just realized that headerPredicate returns as a Java Object representation and not a String representation of the lambda function. 
Googling suggests there is no direct way to stringify it

**Pull request merged**